compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -1068,8 +1069,8 @@ pub trait PrettyPrinter<'tcx>:
             ty::Char if char::try_from(int).is_ok() => {
                 p!(write("{:?}", char::try_from(int).unwrap()))
             }
-            // Raw pointers
-            ty::RawPtr(_) | ty::FnPtr(_) => {
+            // Pointer types
+            ty::Ref(..) | ty::RawPtr(_) | ty::FnPtr(_) => {
```

Comment:
can we really end up here for references? Integer references should only exists for ZSTs, and we shouldn't be printing the address in that case.

Reply:
I am honestly not sure, but in my experience in printing code it's good to have reasonable code paths even for currently impossible cases -- makes debugging a lot easier.